apisyouwonthate.yml

```diff
@@ -72,12 +72,42 @@ rules:
           maxItems: 1
 
   # Author: Andrzej (https://github.com/jerzyn)
-  request-GET-no-body:
-    description: A `GET` request MUST NOT accept a `body` parameter
+  headers-hyphenated-pascal-case:
+    description: All `HTTP` headers MUST use `Hyphenated-Pascal-Case` notation
     severity: error
+    given: "$..parameters[?(@.in == 'header')].name"
+    message: "'HTTP' headers MUST follow 'Hyphenated-Pascal-Case' notation"
     recommended: true
-    given: $.paths..get.parameters..in
+    type: style
+    then: 
+      function: pattern
+      functionOptions:
+        match: "/^([A-Z][a-z0-9]-)*([A-Z][a-z0-9])+/"
+
+  # Author: Andrzej (https://github.com/jerzyn)
+  oas3-hosts-https-only: # checks how does the servers array values start
+    description: "ALL requests MUST go through `https` protocol only"
+    formats:
+      - oas3
+    recommended: true
+    severity: error
+    message: "Servers MUST be https and no other protocol is allowed."
+    given: $.servers..url
+    then:
+      function: pattern
+      functionOptions:
+        match: "/^https:[a-zA-Z0-9./_@-~]+/"
+  
+  # Author: Andrzej (https://github.com/jerzyn)
+  request-support-json-oas3:
+    description: Every request SHOULD support `application/json` media type
+    formats:
+      - oas3
+    recommended: true
+    severity: error
+    message: "{{description}}: {{error}}"
+    given: $.paths.[*].requestBody.content[?(@property.indexOf('json') === -1)]^
     then:
       function: pattern
```

Comment:
Is `pattern` and `falsy` correct here? This is some interesting usage

Reply:
yup, checking whether content always has json for request bodies :)